packages/devtools_app/lib/src/vm_service_wrapper.dart

```diff
@@ -419,14 +423,27 @@ class VmServiceWrapper implements VmService {
     }
   }
 
-  Future<HttpTimelineLoggingState> httpEnableTimelineLogging(
-      String isolateId) async {
+  // TODO(kenz): move this method to
+  // https://github.com/dart-lang/sdk/blob/master/pkg/vm_service/lib/src/dart_io_extensions.dart
+  Future<bool> isHttpLoggingAvailable(String isolateId) async {
+    final Isolate isolate = await getIsolate(isolateId);
+    return isolate.extensionRPCs
+        .contains('ext.dart.io.setHttpEnableTimelineLogging');
```

Comment:
It might be better to generalize this to `areDartIOExtensionsAvailable`, especially since `setHttpEnableTimelineLogging` will always be available if the `dart:io` service extensions are registered.

Reply:
good idea. done.

Reply:
we realized that dart.io extensions can be available even when the timeline logging extensions are not. reverted to checking for timeline logging extensions directly.